restapi/src/main/java/io/stargate/web/docsapi/service/query/FilterExpression.java

```diff
@@ -65,6 +65,7 @@
    * (the worst possible selectivity).
    */
   @Value.Default
+  @Value.Auxiliary // do not use selectivity for equality comparisons
```

Comment:
Why not?

Reply:
Sorry, I put the rationale into git commit message - it's to allow expression simplification in cases where only some of the filter have proper selectivity hints, e.g. `{field: {$eq: 1}, field: {$not: {$ne: 1}}` will not simplify because by default selectivity is `1.0` but inverting the filter makes it `0.0`, so we end up with `{field: {$eq: 1, $selectivity: 1.0}, field: {$eq: 1, $selectivity: 0.0}}`. Those need to compare equal for the simplification rules to kick in.

Reply:
Of course, we could lose user-provided hints this way, but I think it is still better than to miss simplification.

Reply:
Added direct comments about this in the rebased version.